internal/service/networkmanager/core_network_policy_document_data_source.go

```diff
@@ -162,10 +162,6 @@ func DataSourceCoreNetworkPolicyDocument() *schema.Resource {
 									"asn": {
 										Type:     schema.TypeInt,
 										Optional: true,
-										ValidateFunc: validation.Any(
```

Comment:
ASNs are technically `uint32`s, though the AWS API uses `int64`s. We'll have to add a custom `Int64` type to add the validation back